### ISSUE TITLE
it seems the model is initialized with the ground truth point cloud xyz?

### ISSUE THREAD

**Issue by tumble-weed** [sjtuzq/point-radiance#2]
hello,
I had a doubt. In line https://github.com/sjtuzq/point-radiance/blob/501158f58b70ea38b11f06715be8435cb09f8df8/modules/model.py#L40, it seems the the vertex params are directly initialized from the pointcloud? or am i misunderstanding this? the spherical harmonic params however are initialized at random https://github.com/sjtuzq/point-radiance/blob/501158f58b70ea38b11f06715be8435cb09f8df8/modules/model.py#L43

**Comment by sjtuzq** [comment]
There is no ground truth point cloud in the dataset. This point cloud is obtained through a visual hull (sampling the points in the space uniformly and then rejecting those with any of its 2d projections being located in the background mask).

**Comment by tumble-weed** [comment]
i see. the model achieves a recognizable object pretty quickly. i guess the visual hull is pretty informative.

**Comment by haonanhe** [comment]
> There is no ground truth point cloud in the dataset. This point cloud is obtained through a visual hull (sampling the points in the space uniformly and then rejecting those with any of its 2d projections being located in the background mask).

Hi, I want to know how to get the initial point cloud with a visual hull. Is the rejecting process included in
```
105 | pred = images[0, ..., :3].detach().cpu().data.numpy()
106 | gt = self.imagesgt[id].detach().cpu().data.numpy()
107 | # set background as white for visualization
108 | mask = self.masks[id].cpu().data.numpy()
109 | pred = pred*mask+1-mask
110 | gt = gt*mask+1-mask
```
in the main.py by only calculating losses in the mask. Or is the related code not released yet?
